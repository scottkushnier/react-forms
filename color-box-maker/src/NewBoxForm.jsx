
import { useState } from 'react';
  
const initialFormData = {color:'white', height: 30, width: 30};

function NewBoxForm({AddFn}) {
    const [formData, setFormData] = useState(initialFormData);
    const handleChange = (evt) => {
        evt.preventDefault();
        const {name, value} = evt.target;
        setFormData(formData => ({...formData, [name]:value}));
    }
    const handleSubmit = (evt) => {
        evt.preventDefault();
        AddFn(formData);
        setFormData(initialFormData);
    }
    return (
        <form>
            <br/>
            New Block
            <br/>       
            <label htmlFor='color'>Color: </label>
            <input id='color' name='color' value={formData.color} onChange={handleChange}/>   
            <br/>       
            <label htmlFor='height'>Height: </label>
            <input id='height' name='height' value={formData.height} onChange={handleChange}/>
            <br/>       
            <label htmlFor='width'>Width: </label>
            <input id='width' name='width' value={formData.width} onChange={handleChange}/>
            <br/>
            <button onClick={handleSubmit}>Submit</button>
        </form>
    )
}

export default NewBoxForm;