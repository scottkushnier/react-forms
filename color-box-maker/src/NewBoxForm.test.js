

import { render } from "@testing-library/react";
import NewBoxForm from "./NewBoxForm";

test("newboxform smoke test", function () {
  render(<NewBoxForm />)
});

test("matches snapshot", function() {
    const {asFragment} = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
  });
