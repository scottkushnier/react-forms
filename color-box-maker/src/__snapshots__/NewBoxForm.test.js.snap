// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <br />
    New Block
    <br />
    <label
      id="color"
    >
      Color: 
    </label>
    <input
      for="color"
      name="color"
      value="white"
    />
    <br />
    <label
      id="height"
    >
      Height: 
    </label>
    <input
      for="height"
      name="height"
      value="30"
    />
    <br />
    <label
      id="width"
    >
      Width: 
    </label>
    <input
      for="width"
      name="width"
      value="30"
    />
    <br />
    <button>
      Submit
    </button>
  </form>
</DocumentFragment>
`;
